notebooks/script/1.0-pjb-data-exploration.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+get_ipython().run_line_magic('load_ext', 'nb_black')
+
+
+from pathlib import Path
+
+import pandas as pd
+import plotly.express as px
+
+import plotly.io as pio
+
+pio.renderers.default = "png"
+
+
+all_incidents = pd.read_feather(
+    Path("../data/processed/all_incidents.feather")
+).sort_values("current_as_of")
+all_incidents.head(5)
+
+
+# # Basic summary statistics
+# 
+#  - Describe each of the columns
+
+all_incidents.describe(
+    include="all",
+    datetime_is_numeric=True,
+)
+
+
+px.histogram(
+    data_frame=all_incidents.groupby("incident_name").size_acres.max(),
+    x="size_acres",
+    title="Largest Extent (in acres) of Reported Wildfires",
+)
+
+
+start_date_counts = (
+    all_incidents.groupby("incident_name")
+    .date_of_origin.last()
+    .dropna()
+    .dt.date.value_counts()
+)
+
+px.bar(start_date_counts, title="Number of Incidents by Start Date")
+
+
+# # Investigate causes of fires
+# 
+#  - What are all of the causes?
+#  - What are the common groupings of causes?
+#  - How prevalent has each been this season?
+# 
+
+sorted(all_incidents.cause.astype("str").unique().tolist())
+
+
+# fall through ordering
+clean_causes = {
+    "lightning": "lightning",
+    "powerline": "powerline",
+    "shooting": "guns",
+    "arson": "arson",
+    "human": "human",
+    "under invest": "unknown",
+    "unknown": "unknown",
+    "unkown": "unknown",
+    "undetermined": "unknown",
+    "nan": "unknown",
+}
+
+
+def _clean_causes(s):
+    for old, new in clean_causes.items():
+        if old in s.lower():
+            return new
+
+    return s
+
+
+cleaned_causes = all_incidents.cause.astype("str").apply(_clean_causes)
+
+
+px.bar(data_frame=cleaned_causes.value_counts(), title="Count of Causes of Fires")
```

Comment:
I think that we actually want to do a groupby on `incident_name` first, since this will count the same incident multiple times as is.